src/Backends/IF97/IF97Backend.h

```diff
@@ -31,24 +31,62 @@ class IF97Backend : public AbstractState  {
     void set_volu_fractions(const std::vector<CoolPropDbl> &volu_fractions){throw NotImplementedError("Volume composition has not been implemented.");};
     const std::vector<CoolPropDbl> & get_mole_fractions(void){throw NotImplementedError("get_mole_fractions composition has not been implemented.");};
 
-    /// Updating function for incompressible fluid
+    /// Override clear() function of IF97 Water
+    bool clear() {
+    // Reset all instances of CachedElement and overwrite
+    // the internal double values with -_HUGE
+        this->_rhomolar = -_HUGE;
+        this->_T = -_HUGE;
+        this->_p = -_HUGE;
+        this->_Q = -_HUGE;
+        this->_phase = iphase_not_imposed; // Default condition is no phse imposed
+
+        return true;
+    };
+
+    /// Updating function for IF97 Water
     /**
     In this function we take a pair of thermodynamic states, those defined in the input_pairs
-    enumeration and update all the internal variables that we can.
+    enumeration and update the bare minimum values needed to calculate the other values.
 
     @param input_pair Integer key from CoolProp::input_pairs to the two inputs that will be passed to the function
     @param value1 First input value
     @param value2 Second input value
     */
     void update(CoolProp::input_pairs input_pair, double value1, double value2){
+        
+        clear();  //clear the few cached values we are using
+
         switch(input_pair){
             case PT_INPUTS: _p = value1; _T = value2; break;
-            /* TODO:  Add input pairs for saturation functions 
-            case PQ_INPUTS: _p = value1; _Q = value2; break;
-            case QT_INPUTS: _Q = value1; _T = value2; break;
-            */
+            case PQ_INPUTS: 
+                _p = value1; 
+                _Q = value2; 
+                _T = IF97::Tsat97(_p); 
+                if (std::abs(_Q) < 1e-10){
+                    _phase = iphase_liquid; // bubble point
+                }
+                else if (std::abs(value2-1) < 1e-10){
+                    _phase = iphase_gas; // dew point
+                }
+                else
+                    _phase = iphase_twophase;
+                break;
+            case QT_INPUTS: 
+                _Q = value1; 
+                _T = value2; 
+                _p = IF97::psat97(_T); 
+                if (std::abs(_Q) < 1e-10){
+                    _phase = iphase_liquid; // bubble point
```

Comment:
An additional thought from my morning commute...

IF97 needs to be able to stand-alone outside of CoolProp.   This means that IF97 needs to be able to make the phase determination as it determines the IF97Region based on the state.  We shouldn't be doing this within the IF97 Backend, So there are only three states that need to be considered in the IF97 Backend:
- Non-Saturation (iphase_not_imposed): Let IF97 make it's phase determination  
- Q = 0 or 1 (iphase_liquid or iphase_vapor): Call only one IF97 Saturation function  
- 0 < Q < 1 (iphase_two_phase): Call two IF97 Saturation functions and calculate the two phase property (or not).

The 'phase' output variable, if implemented would use the IF97 Region Determination function directly, not just parrot back the '_phase' variable.  Outside the phase envelope, the '_phase' is always iphase_not_imposed to let IF97 do its job. However, if this will cause a conflict in other parts of CoolProp, we can do it another way.

I can always add copious comments as well so that future developers won't be confused by the use of iphase_liquid and iphase_vapor only on the saturation boundary.